tests/requestparsing/basicauth_test.go

```diff
@@ -0,0 +1,117 @@
+package requestparsing
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/google/go-safeweb/testing/requesttesting"
+)
+
+type basicAuthWant struct {
+	headers  map[string][]string
+	ok       bool
+	username string
+	password string
+}
+
+var basicAuthTests = []struct {
+	name    string
+	request []byte
+	want    basicAuthWant
+}{
+	{
+		name: "Basic",
+		request: []byte("GET / HTTP/1.1\r\n" +
+			"Host: localhost:8080\r\n" +
+			"Authorization: Basic UGVsbGU6UGFzc3dvcmQ=\r\n" +
+			"\r\n"),
+		want: basicAuthWant{
+			headers:  map[string][]string{"Authorization": []string{"Basic UGVsbGU6UGFzc3dvcmQ="}},
+			ok:       true,
+			username: "Pelle",
+			password: "Password",
+		},
+	},
+	{
+		name: "NoTrailingEquals",
+		request: []byte("GET / HTTP/1.1\r\n" +
+			"Host: localhost:8080\r\n" +
+			"Authorization: Basic UGVsbGU6UGFzc3dvcmQ\r\n" +
+			"\r\n"),
+		want: basicAuthWant{
+			headers:  map[string][]string{"Authorization": []string{"Basic UGVsbGU6UGFzc3dvcmQ"}},
+			ok:       false,
+			username: "",
+			password: "",
+		},
+	},
+	{
+		name: "DoubleColon",
+		request: []byte("GET / HTTP/1.1\r\n" +
+			"Host: localhost:8080\r\n" +
+			"Authorization: Basic UGVsbGU6UGFzc3dvcmQ6UGFzc3dvcmQ=\r\n" +
+			"\r\n"),
+		want: basicAuthWant{
+			headers:  map[string][]string{"Authorization": []string{"Basic UGVsbGU6UGFzc3dvcmQ6UGFzc3dvcmQ="}},
+			ok:       true,
+			username: "Pelle",
+			password: "Password:Password",
+		},
+	},
+	{
+		name: "NotBasic",
+		request: []byte("GET / HTTP/1.1\r\n" +
+			"Host: localhost:8080\r\n" +
+			"Authorization: xasic UGVsbGU6UGFzc3dvcmQ6UGFzc3dvcmQ=\r\n" +
+			"\r\n"),
+		want: basicAuthWant{
+			headers:  map[string][]string{"Authorization": []string{"xasic UGVsbGU6UGFzc3dvcmQ6UGFzc3dvcmQ="}},
+			ok:       false,
+			username: "",
+			password: "",
+		},
+	},
+	{
+		name: "Ordering",
+		request: []byte("GET / HTTP/1.1\r\n" +
+			"Host: localhost:8080\r\n" +
+			"Authorization: basic QUFBOmFhYQ==\r\n" +
+			"Authorization: basic QkJCOmJiYg==\r\n" +
+			"\r\n"),
```

Comment:
Could you add tests checking whether the ordering is case sensitive? i.e. what if `authorization` is before `Authorization` and the other way around? You can imagine that they're sorted first and then applied ToLower().

I know you made such a test for custom headers, but I wonder if the ones that have some special handling behave differently.